src/ts/background/services/keyring-vault.ts

```diff
@@ -23,6 +27,48 @@ class KeyringVault {
     throw new Error(t('keyringNotInit'))
   }
 
+  getTempPassword (): string {
+    return this._tempPassword || ''
+  }
+
+  setTempPassword (tempPassword: string): void {
+    this._tempPassword = tempPassword
+
+    // Start timer for password expiry
+    this.startExpiryTimer()
+
+    // Trigger clear interval for tempPassword
+    this.clearTempPassword()
+  }
+
+  clearTempPassword (): void {
+    setTimeout(() => {
+      this._tempPassword = undefined
+    }, VALIDITY_INTERVAL * 1000)
+  }
+
+  getExpiryTimeLeft (): number {
+    if (this._expiryTimeLeft < 0 && this._expiryTimeLeftTimerId > 0) {
+      this.clearExpiryTimer()
+    }
+
+    return this._expiryTimeLeft
+  }
+
+  startExpiryTimer (): void {
+    this._expiryTimeLeft = VALIDITY_INTERVAL
+
+    this._expiryTimeLeftTimerId = window.setInterval(() => {
```

Comment:
I wonder if we can use Timeout instead interval here. The benefit of it is that popup client doesn't need call `getExpiryTimeLeft` to clear it.

Reply:
The reason behind using `setInterval` is to update the time left for expiry of either password/mnemonic to display it in the UI. If we use `setTimeout`, it executes only once and not each and every second to update the time left on the UI.

Reply:
i am not referring to clear pwd or mnemonic part. I am talking about the expiry time left timer bit. But i think i get what you mean, it is ok then. 
![image](https://user-images.githubusercontent.com/37568337/63774225-2184d280-c920-11e9-978f-46207e4c7f52.png)

Reply:
Ahh. My bad. For some weird reason, I've observed that the `clearInterval` wasn't being triggered and the `expiryTimeLeft` was going negative. Hence I've added the above check explicitly.